app/src/main/java/io/neurolab/main/output/audio/PlaybackInfoListener.java

```diff
@@ -0,0 +1,51 @@
+package io.neurolab.main.output.audio;
+
+// An abstract class providing some methods(to be implemented) for updating the user interface when media player evokes the corresponding events.
+public abstract class PlaybackInfoListener {
+
+    // Presently we need only 5 basic states. We will add the remaining states as required in the future for our application.
+    static final int INVALID = -1;
+    static final int PLAYING = 0;
+    static final int PAUSED = 1;
+    static final int RESET = 2;
+    static final int COMPLETED = 3;
+
+    public static String convertStateToString(int state) {
+        String stateString;
+        switch (state) {
+            case COMPLETED:
+                stateString = "COMPLETED";
+                break;
+            case INVALID:
+                stateString = "INVALID";
+                break;
+            case PAUSED:
+                stateString = "PAUSED";
+                break;
+            case PLAYING:
+                stateString = "PLAYING";
+                break;
+            case RESET:
+                stateString = "RESET";
+                break;
+            default:
+                stateString = "N/A";
+        }
+        return stateString;
+    }
+
+    void onLogUpdated(String formattedMessage) {
```

Comment:
These are abstract methods right? We'll use correct syntax for them. 
`public abstract void onLogUpdated(String formattedMessage);`

Reply:
ahh.. kept it for later and missed that. thanks. have updated.

Reply:
These should be abstract methods right? Codacy also giving errors on these